server/routes/wishCards.js

```diff
@@ -80,89 +102,102 @@ const AgencyRepository = require('../db/repository/AgencyRepository');
+      }
     }
-  }
-});
+  },
+);
 
 // @desc    wishcard creation form (guided process, i.e. uses default toy choices) sends data to db
 // @route   POST '/wishcards/guided/'
-// @access  Private, must be verified as a partner
+// @access  Private, verified partners
 // @tested 	Yes
-router.post('/guided/', upload.single('wishCardImage'), async (req, res) => {
-  if (req.file === undefined) {
-    handleError(
-      res,
-      400,
-      `Error: File must be in jpeg, jpg, gif, or png format. The file
+router.post(
+  '/guided/',
+  checkPermissions,
+  upload.single('wishCardImage'),
+  createGuidedWishcardValidationRules(),
+  validate,
+  async (req, res) => {
+    if (req.file === undefined) {
+      handleError(
+        res,
+        400,
+        `Error: File must be in jpeg, jpg, gif, or png format. The file
         mst also be less than 5 megabytes.`,
-    );
-  } else {
-    try {
-      let { itemChoice } = req.body;
-      itemChoice = JSON.parse(itemChoice);
-      await WishCardRepository.createNewWishCard({
-        childBirthday: new Date(req.body.childBirthday),
-        wishItemName: itemChoice.Name,
-        wishItemPrice: Number(itemChoice.Price),
-        wishItemURL: itemChoice.ItemURL,
-        wishCardImage: req.file.location,
-        createdBy: res.locals.user._id,
-        address: {
-          address1: req.body.address1,
-          address2: req.body.address2,
-          city: req.body.address_city,
-          state: req.body.address_state,
-          zip: req.body.address_zip,
-          country: req.body.address_country,
-        },
-        ...req.body,
-      });
-
-      res.status(200).send('/wishcards/');
-    } catch (error) {
-      handleError(res, 400, error);
+      );
+    } else {
+      try {
+        let { itemChoice } = req.body;
+        itemChoice = JSON.parse(itemChoice);
+        await WishCardRepository.createNewWishCard({
+          childBirthday: new Date(req.body.childBirthday),
+          wishItemName: itemChoice.Name,
+          wishItemPrice: Number(itemChoice.Price),
+          wishItemURL: itemChoice.ItemURL,
+          wishCardImage: req.file.location,
+          createdBy: res.locals.user._id,
+          address: {
+            address1: req.body.address1,
+            address2: req.body.address2,
+            city: req.body.address_city,
+            state: req.body.address_state,
+            zip: req.body.address_zip,
+            country: req.body.address_country,
+          },
+          ...req.body,
+        });
+
+        res.status(200).send({ success: true, url: '/wishcards/' });
+      } catch (error) {
+        handleError(res, 400, error);
+      }
     }
-  }
-});
+  },
+);
 
 // @desc    Render wishCards.html which will show all wishcards
 // @route   GET '/wishcards'
-// @access  Public, all users can see
+// @access  Private, all users can see
 // @tested 	Yes
-router.get('/', async (_req, res) => {
+router.get('/', redirectLogin, async (_req, res) => {
```

Comment:
doesnt this prevent everyone from seeing wishcards who is not logged in

Reply:
Yeah, true. Changed that back; changed tests as well.